### PR TITLE
[SPARK-23698] Remove raw_input() from Python 2

### PR DESCRIPTION
**Pull request by cclauss**
Signed-off-by: cclauss <cclauss@bluewin.ch>

## What changes were proposed in this pull request?

Humans will be able to enter text in Python 3 prompts which they can not do today.
The Python builtin __raw_input()__ was removed in Python 3 in favor of __input()__.  This PR does the same thing in Python 2.

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)
(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)
flake8 testing

Please review http://spark.apache.org/contributing.html before opening a pull request.

### CODE REVIEW

File: dev/create-release/releaseutils.py [hunk 1]
```diff
@@ -49,13 +49,16 @@
     print("Install using 'sudo pip install unidecode'")
     sys.exit(-1)
 
+if sys.version < '3':
+    input = raw_input
```
Comment (HyukjinKwon) [inline dev/create-release/releaseutils.py]: If we can do the opposite, the diff should be only 4 lines though

**Review by HyukjinKwon — COMMENTED** [review]
Are you actually able to roughly check if those scripts can be ran against Python 3?

File: dev/create-release/releaseutils.py (hunk 1 above)
Comment (cclauss) [reply dev/create-release/releaseutils.py]: Two downsides to that approach:
1. We stick with the legacy Python syntax which will unnecessarily complicate our lives (and our diffs) [in 18 months](http://pythonclock.org)
2. This approach reduces the linting errors from 10 down to just 2.

**Comment by cclauss** [comment]
Tested data entry...
In __./dev/merge_spark_pr.py__ just after __clean_up()__, I added the lines:
```
while not continue_maybe('y to conntinue'):
    print('loop')
sys.exit()
```
Test: 'y' or 'Y' caused a loop while all others including "", "yes", "n", "N", "0", "1", "." caused an exit()
Identical results on both Python 2 and Python 3
---
In __./dev/create-release/releaseutils.py__ just after __yesOrNoPrompt()_ I added the lines:
```
while not yesOrNoPrompt('y to quit'):
    print('got an 'n'')
sys.exit()
```
Test: 'y' caused an exit() while all others including "", "Y", "yes", "n", "N", "0", "1", "." caused a loop
Identical results on both Python 2 and Python 3

**Comment by HyukjinKwon** [comment]
ok to test

**Review by HyukjinKwon — APPROVED** [review]
LGTM

**Comment by HyukjinKwon** [comment]
Merged to master

**Pull request closed without merging**